runtime/lua/vim/lsp/util.lua

```diff
@@ -525,26 +525,57 @@ function M.make_floating_popup_options(width, height, opts)
   }
 end
 
+local function _add_tagstack(item)
+  local function should_add(new_item)
+    local stack = vim.fn.gettagstack()
+
+    -- Check if we're at the bottom of the tagstack.
+    if stack.curidx <= 1 then return true end
+
+    local top_item = stack.items[stack.curidx-1]
+
+    -- Check if the item at the top of the tagstack is exactly the
+    -- same as the one we want to push.
+    if top_item.tagname ~= new_item.tagname then return true end
+    for i, v in ipairs(top_item.from) do
+      if v ~= new_item.from[i] then return true end
+    end
+    return false
+  end
```

Comment:
Why is this a function here?

You could either define it outside of the above function or just make this inlined, right? I don't currently see why it's written this way.

Reply:
(I just now see the comment you left earlier @h-michael . I have left a comment there now as well).

Reply:
I don't know Lua well enough to justify this, unfortunately. There is overhead by declaring it like I did, right? Or does it get inlined somehow?

Reply:
You know what, it was weird to have that function declared inside another one. Now it's a single, private function that checks whether item should be added to tagstack. :smiley: